### PR TITLE
continue opt175b download from the latest available file

### PR DESCRIPTION
**Pull request by Barqawiz**
**Patch Description**
Add script to continue the download after the latest available opt175b file in the case of an internet issue or interruption.

**Testing steps**
The following cases are tested:
- file does not exist.
- file exist with zero byte in case of running the script without internet.

related request #288

### CODE REVIEW

File: metaseq/scripts/download_checkpoint_opt175b.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  do
+      filename="checkpoint_last-model_part-$part_id-shard$shard_id.pt"
+
+      if [ ! -s $filename ]; then
+        wget -O "$2$filename" "${presigned_url/$str_to_replace/$filename}"
+      else
+        echo "- ${filename} exist"
+      fi
+  done
+done
```
Comment (stephenroller) [inline metaseq/scripts/download_checkpoint_opt175b.sh:27]:
```suggestion
done

```

**Review by stephenroller — COMMENTED** [review]
leaving this one for @suchenzang to review but lgtm

**Comment by suchenzang** [comment]
What's the reason for committing a new script instead of modifying metaseq/scripts/download_opt175b.sh ?

**Comment by Barqawiz** [comment]
@suchenzang, to keep the choice of downloading everything from scratch if needed or using the new script to download from the last point. For any reason, if you want to merge it? I can do it. Otherwise, we can keep both options.

**Comment by suchenzang** [comment]
Got it - yes, it would be cleaner to control through a flag from one file. And update documentation accordingly to highlight the two use-cases / ways to invoke the script (download from scratch vs from last saved).

**Comment by Barqawiz** [comment]
ok sure will do it.

**Comment by Barqawiz** [comment]
There are two parameters at the moment $1 mandatory for presigned_url and $2 optional for the directory. 

The available options to add a third optional parameter:
- **Option1**: keep the mandatory parameter $1 and add flags for both optional parameters regardless of the order. (-d for directory -c for checkpoint).
- **Option2**: move the directory parameter to $3 and use $2 for the checkpoint flag. 
- **Option3**: use $3 for the checkpoint flag and keep $2 for the directory. The issue of this option, the directory, should always be mentioned before the flag.

@suchenzang kindly advise which option is more suitable to start the implementation.

**Comment by suchenzang** [comment]
@Barqawiz Option 1! Thanks for adding flags - it's definitely better than positional args :)

**Comment by Barqawiz** [comment]
@suchenzang the merge is ready for your review.

The expected format, -c means continue download:
bash ./download_opt175b.sh "<presigned_url_given_in_email>" -d "<target directory>" -c

Also I added a validation if the user did not enter the presigned_url.

**Review by suchenzang — APPROVED** [review]
Amazing, thank you!!

**Pull request merged**